### PR TITLE
Add wallmounted support for plantlike and plantlike_rooted nodes

### PR DESCRIPTION
**Pull request by Wuzzy2**
This PR adds wallmounted support for plantlike and plantlike\_rooted.

https://satoshiupload.com/images/IkrdJ0QggH.png
https://satoshiupload.com/images/inVgwyU1rP.png
https://satoshiupload.com/images/SKaf2dOXs0.png

## How to test

Use the new wallmounted plantlike nodes in DevTest in combination with the Param2 tool.

## Possibly competing PRs

#11312 (facedir)

### CODE REVIEW

**Comment by celeron55** [comment]
These look weird, mainly because IRL plants growing from walls almost always grow directly upwards against gravity or directly downwards along gravity. These are at a 45° angle to both of those. Is there anything that would look natural when represented with this drawtype?

**Comment by Wuzzy2** [comment]
* Corals
* Crystals
* Roots from above

File: src/client/content_mapblock.cpp [hunk 1]
```diff
@@ -998,6 +1026,22 @@ void MapblockMeshGenerator::drawPlantlike()
 		break;
 	}
 
+	if (is_rooted) {
```
Comment (sfan5) [inline src/client/content_mapblock.cpp:1029]: If wallmounted is also supposed to work for plantlike alone why do this?

Comment (celeron55) [reply src/client/content_mapblock.cpp:1029]: It seems to work so it can't be incorrect.

**Review by celeron55 — APPROVED** [review]

**Comment by Wuzzy2** [comment]
@sfan5: The `is_rooted` is done because the "plant" in `plantlike_rooted` needs to be offset. Actually, wallmounted for `plantlike_rooted` works a little bit differently. The base cube stays in place, but the "plant" changes the side, i.e. normally it is on the top side of the node, but with wallmounted it can "grow" to the left, right, front, back or bottom. See the screenshots.

I noticed the documentation on `plantlike_rooted` could have been a little more explicit, so I just updated it.

**Review by SmallJoker — APPROVED** [review]
Works

**Comment by SmallJoker** [comment]
Õ.Õ lint please...
```
Checking builtin/game/falling.lua                 1 warning

    builtin/game/falling.lua:145:72: empty if branch
```

**Comment by Wuzzy2** [comment]
OK, done.

**Pull request merged**